servers/braze-content-proxy/src/routes/digest.ts

```diff
@@ -40,16 +40,17 @@ function transformToBrazePayload(
     return [];
   }
   const savedItems: BrazeSavedItem[] = edges
-    .filter((edge) => edge.node.item.__typename === 'Item')
+    .filter(
+      (edge) =>
+        edge?.node?.item != null && edge?.node?.item.__typename === 'Item',
+    )
     .map((edge) => {
-      const item = edge.node.item;
-      if (item.__typename === 'Item') {
-        return {
-          title: item.preview.title,
-          imageUrl: item.preview.image?.cachedImages[0].url,
-          url: item.preview.url,
-        };
-      }
+      const item = edge?.node?.item as Item; // force cast to Item because map can't infer type from the above filter.
```

Comment:
you can use a type predicate in the filter function if you want. it's not too different to what you're doing here but just moves the hint up

```
  .filter(
      (edge): edge is Item =>
        edge?.node?.item != null && edge?.node?.item.__typename === 'Item',
    )
    ```